staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -455,6 +455,17 @@ func (s *sharedIndexInformer) SetTransform(handler TransformFunc) error {
 	return nil
 }
 
+func (s *sharedIndexInformer) SetStore(store ThreadSafeStore) error {
```

Comment:
I think it is better to construct the right thing from the start than to side-effect an informer after it is constructed.  I recommend extending SharedIndexInformerOptions with the option to specify the Store.